src/analysis/souffle/examples/multimic/multimic-preferences.authlogic

```diff
@@ -0,0 +1,335 @@
+  // be done with UserC's data and to transform this data in any way.
+  "AmbientPrivacyPolicy" canSay PrincipalX may(ActionX, TagX) :-
+    isAction(ActionX), isPrincipal(PrincipalX), isTag(TagX).
+  "AmbientPrivacyPolicy" canSay downgrades(PathX, TagX) :-
+    isPath(PathX), isTag(TagX).
+  "AmbientPrivacyPolicy" canSay hasTag(PathX, TagX) :-
+    isPath(PathX), isTag(TagX).
+  
+  // This is stuff that needs to be generated behind the scenes
+  // (Clearly C's policy should have no knowledge of this)
+  isAction("deliver_privacy_notifications").
+  isAction("local_computation").
+  isAction("store").
+  isPrincipal("AmbientPrivacyPolicy").
+  isPrincipal("Notifier").
+  isPrincipal("MicCompute").
+  isPrincipal("SpeechToTextStorageSink").
+  isPath("audioIn").
+  isPath("asrStorageConn").
+  isPath("speechToTextIn").
+  isTag("UserC_userSpeech").
+  isTag("UserC_speechTranscript").
+}
+
+// The facts here are produced by some part of the microphone's operating 
+// system. This module is trusted by the "AmbientPrivacyPolicy"
+// to correctly determine which user owns the device.
+"DeviceRoleAuthority" says {
+  "UserA" ownsDevice("this_microphone").
+}
+
+// This is the (presumably open-source) privacy policy "EULA" that users 
+// delegate to.
+"AmbientPrivacyPolicy" says {
+
+  // This policy allows end-users to express their own preferences and state 
+  // their own preference for how dispute resolution should be done. 
+  UserX canSay UserX hasPrivacyPreference("audio", PreferenceX) :-
+    isPrivacyPreference(PreferenceX), isPrincipal(UserX).
+  UserX canSay UserX setsPrivacyResolution("audio", ResolutionX) :-
+    isPrivacyResolution(ResolutionX), isPrincipal(UserX).
+
+  "DeviceRoleAuthority" canSay UserX ownsDevice("this_microphone") :-
+    isPrincipal(UserX).
+
+  // The privacy preferences of some members of the room might not be met if 
+  // the owner of the device says that it needs to stay on.
+  overridePreferences("audio") :- UserX ownsDevice("this_microphone"),
+    UserX setsPrivacyResolution("audio", "stay_on").
+
+  // Regardless of preferences, notification that there is a microphone is 
+  // always allowed.
+  "Notifier" may("deliver_privacy_notifications", UserX_userSpeech) :-
+    isUserSpeech(UserX_userSpeech).
+
+  // The pipleine involving transcription and storage of the transcript is 
+  // allowed to run for a particular user if:
+  // 1) the user's preference is either permissive or that just transcripts can
+  // be stored.
+  // 2) the device owner's preference implies we should override the 
+  // preferences of others
+  // The rule "allow_transcription" captures these alternatives:
+  allowTranscription(UserX) :-
+    UserX hasPrivacyPreference("audio", "only_store_transcript").
+  allowTranscription(UserX) :- UserX hasPrivacyPreference("audio", "permissive").
+  allowTranscription(UserX) :- overridePreferences("audio"), isPrincipal(UserX).
+
+  "MicCompute" may("local_computation", UserX_userSpeech) :-
```

Comment:
Because [the implementation of the may/will check ](https://github.com/google-research/raksha/pull/209#pullrequestreview-803727127) works by checking if the original data owner says something `may` happen, the only way these rules could  lead to the application being accepted (in other words prove that the `will` part is allowed) is if the user delegated to the `"AmbientPrivacyPolicy"` (which in this example the users do -- for example on lines `47-52` is where UserA does this).